apps/condo/domains/meter/tasks/sendSubmitMeterReadingsPushNotifications.js

```diff
@@ -112,42 +146,89 @@ const sendSubmitMeterReadingsPushNotifications = async () => {
             offset: state.offset,
             pageSize: state.pageSize,
         })
-        const meterReadings = await readMeterReadings({ context, meters })
         const organizations = await readOrganizations({ context, meters })
+        const reportingPeriods = await readMeterReportingPeriods({ context, organizations })
+        const meterReadings = await readMeterReadings({ context, meters, reportingPeriods })
+
+        const metersWithoutReadings = []
+        const periodsByProperty = []
+        const periodsByOrganization = []
+        let defaultPeriod = null
+
+        for (let period of reportingPeriods) {
+            if (!period.organization) defaultPeriod = period
+            else if (!period.property) periodsByOrganization.push(period)
+            else periodsByProperty.push(period)
+        }
+
+        for (const meter of meters) {
+
+
+
+            if (meter.deletedAt) console.log('deletedMeter123', meter)
+
+
 
-        // right join data to metersPage
-        const metersWithoutReadings = rightJoin(
-            meters,
-            meterReadings,
-            (meter, item) => meter.id === item.meter.id,
-            (meter, readings) => ({ meter, readings })
-        )
-            .filter(({ readings }) => readings.length === 0)
-            .map(({ meter }) => meter)
+
+            const period = periodsByProperty.find(({ property }) => property.id === meter.property.id) ??
+                periodsByOrganization.find(({ organization }) => organization.id === meter.organization.id) ??
+                defaultPeriod
+
+            if (isNil(period)) continue
+
+            const notifyStartDay = get(period, 'notifyStartDay')
+            const notifyEndDay = get(period, 'notifyEndDay')
+
+            const readingsOfCurrentMeter = compact(meterReadings.map(reading => {
```

Comment:
readingsOfCurrentMeter is that correct that some items going to be undefined?
Maybe just use .filter(conditionFromIfStatement)?